libsolidity/interface/CompilerStack.cpp

```diff
@@ -194,6 +203,16 @@ bool CompilerStack::parse()
 				else
 					noErrors = false;
 
+				if (m_contracts.find(contract->name()) != m_contracts.end())
+				{
+					const ContractDefinition* existingContract = m_contracts.find(contract->name())->second.contract;
+					if (contract != existingContract)
+						BOOST_THROW_EXCEPTION(CompilerError() <<
+							errinfo_sourceLocation(contract->location()) <<
+							errinfo_comment(contract->name() + " conflicts with!!! contract at "
```

Comment:
Yeah, while I was sorting some test case stuff out.  I can't believe I didn't see that.  No more commits before bed for me.